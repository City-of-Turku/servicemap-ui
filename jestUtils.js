import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });