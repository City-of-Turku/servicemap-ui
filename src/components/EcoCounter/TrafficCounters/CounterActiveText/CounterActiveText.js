import React from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import { Typography } from '@mui/material';
import styled from '@emotion/styled';
import { formatFullDates } from '../../utils';

/**
 * Render text that shows from and until dates.
 * @prop {string} dataFrom
 * @prop {string} dataUntil
 * @returns JSX eement
 */
const CounterActiveText = ({ dataFrom, dataUntil }) => {
  const intl = useIntl();
  const dataFromFormat = formatFullDates(dataFrom);
  const dataUntilFormat = formatFullDates(dataUntil);

  return (
    <StyledTextContainer>
      <Typography variant="body2" sx={{ mb: '0.4rem' }}>
        {intl.formatMessage(
          { id: 'ecocounter.station.counts.period' },
          { value1: dataFromFormat, value2: dataUntilFormat },
        )}
      </Typography>
    </StyledTextContainer>
  );
};

const StyledTextContainer = styled.div(({ theme }) => ({
  textAlign: 'center',
  margin: `${theme.spacing(1)} 0`,
}));

CounterActiveText.propTypes = {
  dataFrom: PropTypes.string,
  dataUntil: PropTypes.string,
};

CounterActiveText.defaultProps = {
  dataFrom: '',
  dataUntil: '',
};

export default CounterActiveText;
