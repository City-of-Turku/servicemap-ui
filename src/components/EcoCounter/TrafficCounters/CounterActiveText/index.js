import CounterActiveText from './CounterActiveText';

export default CounterActiveText;
