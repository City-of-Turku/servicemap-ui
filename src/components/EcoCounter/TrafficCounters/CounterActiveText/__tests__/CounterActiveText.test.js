/* eslint-disable react/jsx-props-no-spreading */
// Link.react.test.js
import React from 'react';
import CounterActiveText from '../index';
import { getRenderWithProviders } from '../../../../../../jestUtils';

const mockProps = {
  dataFrom: '2020-01-01',
  dataUntil: '2023-10-10',
};

const renderWithProviders = getRenderWithProviders({});

describe('<CounterActiveText />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<CounterActiveText {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<CounterActiveText {...mockProps} />);

    const p = container.querySelectorAll('p');
    expect(p[0].textContent).toEqual('Laskentatiedot ovat väliltä 01.01.2020 - 10.10.2023');
  });
});
