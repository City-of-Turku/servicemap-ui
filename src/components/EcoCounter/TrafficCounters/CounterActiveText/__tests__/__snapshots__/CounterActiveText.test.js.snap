// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CounterActiveText /> should work 1`] = `
<div>
  <div
    class="css-aavt1n"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 css-1bvqycf-MuiTypography-root"
    >
      Laskentatiedot ovat väliltä 01.01.2020 - 10.10.2023
    </p>
  </div>
</div>
`;
